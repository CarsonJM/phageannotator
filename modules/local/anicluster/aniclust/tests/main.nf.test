nextflow_process {

    name "Test process: ANICLUSTER_ANICLUST"
    script "../main.nf"
    process "ANICLUSTER_ANICLUST"
    tag "modules"
    tag "modules_local"
    tag "anicluster"
    tag "anicluster_aniclust"


    test("['bacteroides_fragilis']['illumina']['contigs_fa_gz'] & ['modules_local']['ani_tsv']") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['bacteroides_fragilis']['illumina']['contigs_fa_gz'], checkIfExists: true ) ],
                                                [ file(params.test_data['modules_local']['ani_tsv'], checkIfExists: true ) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
