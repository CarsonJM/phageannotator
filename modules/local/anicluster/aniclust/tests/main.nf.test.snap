{
    "['bacteroides_fragilis']['illumina']['contigs_fa_gz'] & ['modules_local']['ani_tsv']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,adcc2d4d4b096d6c9640180e94dd6c5f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ],
                "clusters": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,adcc2d4d4b096d6c9640180e94dd6c5f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ]
            }
        ],
        "timestamp": "2023-11-13T11:49:27.609639159"
    }
}